### PR TITLE
Create additional nsgroup objects

### PR DESCRIPTION
**Pull request by badnetmask**
As required for https://github.com/infobloxopen/infoblox-ansible/pull/56, I patched this app to add support for the additional nsgroup objects.

However, I found a couple of problems during my build, which may need some further attention before merging:

1. During image build, there was an error to compile the `cryptography` Python package, which required installing `cargo` via `apk`.
2. The `rust` version installed on Alpine 3.6 is too old, so I updated the Dockerfile to use 3.14 instead.

Please, let me know your thoughts about this.

(CC: @anagha-infoblox)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,8 +1,8 @@
-FROM alpine:3.6
+FROM alpine:3.14
 
 ADD flaskapp.py /root/flaskapp.py
 ADD requirements.txt /root/requirements.txt
-RUN apk add --no-cache build-base openssl-dev libffi-dev python3 python3-dev py3-pip
+RUN apk add --no-cache build-base openssl-dev libffi-dev python3 python3-dev py3-pip cargo
```
Comment (mattclay) [inline Dockerfile]: Try this instead, it results in a faster build and a smaller container:

```suggestion
RUN apk add --no-cache build-base python3 py3-wheel py3-pip py3-cryptography
```

File: flaskapp.py [hunk 2]
```diff
@@ -266,7 +274,11 @@ def __init__(self):
             'record:mx': [],
             'record:srv': [],
             'record:naptr': [],
-            'record:txt': []
+            'record:txt': [],
+            'nsgroup:delegation': [],
+            'nsgroup:forwardingmember': [],
+            'nsgroup:forwardstubserver': [],
+            'nsgroup:stubmember': []
```
Comment (mattclay) [inline flaskapp.py]: Leave a trailing comma for cleaner diffs in the future:

```suggestion
            'nsgroup:stubmember': [],
```

File: flaskapp.py [hunk 3]
```diff
@@ -230,7 +234,11 @@ def to_dict(self, fields=[]):
             'zone_format': self.zone_format,
             'grid_primary': self.grid_primary,
             'grid_secondaries': self.grid_secondaries,
-            'ns_group': self.ns_group
+            'ns_group': self.ns_group,
+            'delegate_to': self.delegate_to,
+            'forwarding_servers': self.forwarding_servers,
+            'external_servers': self.external_servers,
+            'stub_members': self.stub_members
```
Comment (mattclay) [inline flaskapp.py]: Leave a trailing comma for cleaner diffs in the future:

```suggestion
            'stub_members': self.stub_members,
```

File: Dockerfile [hunk 4]
```diff
@@ -1,8 +1,8 @@
-FROM alpine:3.6
+FROM alpine:3.14
```
Comment (mattclay) [inline Dockerfile]: 3.15 is available now. Since the base container is being updated, it might be good to use that version instead.

**Review by mattclay — CHANGES_REQUESTED** [review]

**Review by mattclay — COMMENTED** [review]
@badnetmask Have you verified this updated container works with all of the existing infoblox collection tests, as well as the changes in https://github.com/infobloxopen/infoblox-ansible/pull/56?

**Comment by badnetmask** [comment]
I just ran all the integration tests on my local machine, using my branch created for the PR. All passed just fine.

**Review by mattclay — APPROVED** [review]

**Pull request merged**